pkg/sql/parser/sql.y

```diff
@@ -5983,13 +5962,45 @@ set_csetting_stmt:
 
 // %Help: ALTER TENANT - alter tenant configuration
 // %Category: Group
+// %SeeAlso: ALTER TENANT REPLICATION, ALTER TENANT CLUSTER SETTING
+alter_tenant_stmt:
+  alter_tenant_replication_stmt // EXTEND WITH HELP: ALTER TENANT REPLICATION
+| alter_tenant_csetting_stmt    // EXTEND WITH HELP: ALTER TENANT CLUSTER SETTING
+| ALTER TENANT error            // SHOW HELP: ALTER TENANT
+
+// %Help: ALTER TENANT REPLICATION - alter tenant replication stream
+// %Category: Group
+// %Text:
+// ALTER TENANT '<tenant_name>' PAUSE REPLICATION
+// ALTER TENANT '<tenant_name>' RESUME REPLICATION
+alter_tenant_replication_stmt:
+  ALTER TENANT sconst_or_placeholder PAUSE REPLICATION
```

Comment:
@knz / @dt or anyone else who better understands the rules we have to accept an expression, is `sconst_or_placeholder` the correct rule to accept a tenant name. Currently, `CREATE TENANT` accepts a `name` but we want to change that as well so that we can support placeholders, Additionally, we want to reduce the possibility of us misquoting these names when storing them in the system table (an error I already made in another PR and has been fixed since).